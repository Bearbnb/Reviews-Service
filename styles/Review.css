
.review {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.mediaobject {
    border-radius: 100px
}