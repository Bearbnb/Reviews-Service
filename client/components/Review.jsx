import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import ReviewWithoutComments from './ReviewWithoutComments.jsx';
import HostComents from './HostComments.jsx';
import styles from '../../styles/Review.css';

const Review = ({ review, host, FlagClick }) => {
  if (review.host_comments === 'null') {
    return (
      <div>
        <ReviewWithoutComments review={review} FlagClick={FlagClick} />
        <div className={styles.bottom} />
      </div>
    );
  }
  return (
    <div>
      <ReviewWithoutComments review={review} FlagClick={FlagClick} />
      <HostComents review={review} host={host} />
      <div className={styles.bottom} />
    </div>);
};

Review.propTypes = {
  review: PropTypes.shape({
    host_comments: PropTypes.string.isRequired,
    created: PropTypes.string.isRequired,
    house_id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    photo: PropTypes.string.isRequired,
    rating: PropTypes.array.isRequired,
    review: PropTypes.string.isRequired,
  }).isRequired,
  host: PropTypes.shape({
    name: PropTypes.string,
    created: PropTypes.string,
    photo: PropTypes.string,
  }).isRequired,
  FlagClick: PropTypes.func.isRequired,
};
export default Review;
