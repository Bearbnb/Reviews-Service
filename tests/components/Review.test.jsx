import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Review from '../../client/components/Review';

describe('<Review />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Review review={{
      host: { photo: '', host_comments: '' },
      review: {},
    }}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});