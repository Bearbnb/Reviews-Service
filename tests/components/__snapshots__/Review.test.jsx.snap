// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> should render properly 1`] = `
<div>
  <ReviewWithoutComments
    review={
      Object {
        "FlagClick": true,
        "host": Object {
          "host_comments": "",
          "photo": "",
        },
        "review": Object {
          "created": "",
        },
      }
    }
  />
  <HostComments
    host={[Function]}
    review={
      Object {
        "FlagClick": true,
        "host": Object {
          "host_comments": "",
          "photo": "",
        },
        "review": Object {
          "created": "",
        },
      }
    }
  />
  <div
    className="bottom"
  />
</div>
`;
